ipa-core/src/helpers/transport/mod.rs

```diff
@@ -36,14 +40,30 @@ use crate::{
 
 /// An identity of a peer that can be communicated with using [`Transport`]. There are currently two
 /// types of peers - helpers and shards.
-pub trait Identity: Copy + Clone + Debug + PartialEq + Eq + Hash + Send + Sync + 'static {}
+pub trait Identity:
+    Copy + Clone + Debug + PartialEq + Eq + PartialOrd + Ord + Hash + Send + Sync + 'static
+{
+    fn as_str<'a>(&self) -> Cow<'a, str>;
```

Comment:
Is the lifetime parameter useful here? Seems like returning `Cow<'static, str>` would be fine. If support for borrowing it from `self` is important, then the lifetime parameter should be attached to `&self`.